src/core_plugins/timelion/server/series_functions/es/lib/build_request.js

```diff
@@ -77,4 +76,11 @@ export default function buildRequest(config, tlConfig, scriptedFields) {
       size: 0
     }
   };
+
+  const timeout = tlConfig.server.config().get('elasticsearch.shardTimeout');
+  if (timeout) {
```

Comment:
Are we sure that timeout is a number? because if the server config return a string containing `'0'` this will ends up adding the timeout

Reply:
Very good question :-) @spalger does the Joi validation (https://github.com/elastic/kibana/blob/master/src/core_plugins/elasticsearch/index.js#L53) if it's set to number make sure the data type here will be coming in as a number?

Reply:
I've done some testing and seems to validate and return always a number. If not a number it will throw errors.